cloud/services/object_storage_buckets.go

```diff
@@ -0,0 +1,153 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/linode/linodego"
+
+	"github.com/linode/cluster-api-provider-linode/cloud/scope"
+)
+
+func EnsureObjectStorageBucket(ctx context.Context, bucketScope *scope.ObjectStorageBucketScope, logger logr.Logger) (*linodego.ObjectStorageBucket, error) {
```

Comment:
Nevermind, it would appear logger injection isn't a thing in controller-runtime (any more), and the recommendation is to either use `sigs.k8s.io/controller-runtime.Log` for getting the root-level logger, or storing a logger in the reconciler's field.